cranelift/codegen/src/isa/pulley_shared/abi.rs

```diff
@@ -70,6 +70,19 @@ where
         let mut next_v_reg = 0;
         let mut next_stack: u32 = 0;
 
+        let ret_area_ptr = if add_ret_area_ptr {
+            assert!(ArgsOrRets::Args == args_or_rets);
+            next_x_reg += 1;
+            Some(ABIArg::reg(
+                x_reg(0).to_real_reg().unwrap(),
+                I64,
+                ir::ArgumentExtension::None,
+                ir::ArgumentPurpose::Normal,
+            ))
```

Comment:
To make this code a little future proof to shuffling it around at some point in the future, lets not hard code `0`:

```suggestion
            next_x_reg += 1;
            Some(ABIArg::reg(
                x_reg(next_x_reg - 1).to_real_reg().unwrap(),
                I64,
                ir::ArgumentExtension::None,
                ir::ArgumentPurpose::Normal,
            ))
```

It may be convenient to have a helper closure that does the increment of `next_x_reg` and returns `Option<Reg>` or whatever for you. I'll leave it up to you, and how much you are annoyed by `next_x_reg - 1` :-p